File: app/src/test/java/de/jodamob/kotlin/droidcon/d/RadioState2Test.java
package de.jodamob.kotlin.droidcon.d;

import org.junit.Test;

import static org.junit.Assert.assertFalse;

public class RadioState2Test {

    @Test
    public void should_be_off() {
        RadioState tested = RadioState.ALL_OFF;
        assertFalse(tested.isCellularNetworkConnected());
        assertFalse(tested.isWifiNetworkConnected());
    }
}

File: app/src/test/java/de/jodamob/kotlin/droidcon/d/RadioState2Test.kt
package de.jodamob.kotlin.droidcon.d

import org.junit.Assert.assertFalse
import org.junit.Test

class RadioState2Test {

    @Test
    fun should_be_off() {
        val tested = RadioState.ALL_OFF
        tested.assertNotOnAnyNetwork()
    }

    private fun RadioState.assertNotOnAnyNetwork() {
        assertFalse(isOnAnyNetwork())
    }

    private fun RadioState.isOnAnyNetwork() : Boolean {
        return isCellularNetworkConnected && isWifiNetworkConnected
    }
}